23-input-scanning/exercises/04-grep/main.go

```diff
@@ -34,4 +41,29 @@ package main
 // ---------------------------------------------------------
 
 func main() {
+	var (
+		in      = bufio.NewScanner(os.Stdin)
+		lineMap = map[string]bool{}
+		query   = os.Args[1]
+	)
+
+	in.Split(bufio.ScanLines)
+	for in.Scan() {
+		line := in.Text()
+		_, ok := lineMap[line]
+
+		if ok {
+			fmt.Println(line)
+			return
+		}
+
+		a := strings.Fields(line)
```

Comment:
a is not a good name; it does not tell me anything, use a more meaningful name like words etc

Reply:
fixed and changed to lineSlice